src/entities/test-Result.entity.ts

```diff
@@ -0,0 +1,51 @@
+import {
+    Entity,
+    PrimaryGeneratedColumn,
+    Column,
+    CreateDateColumn,
+    UpdateDateColumn,
+    DeleteDateColumn,
+    JoinColumn,
+    ManyToOne,
+} from "typeorm";
+import { TestSample, LabStaff } from "@/entities"
+
+@Entity({ name: "test_result"})
+export class TestResult {
+    @PrimaryGeneratedColumn("uuid")
+    id!: string;
+
+    @Column({ type: "boolean", default: false, name: "test_completed", nullable: false })
+    testCompleted!: boolean;
+
+    @Column({ type: "jsonb", name: "sample_result", nullable: false })
+    sampleResult!: Record<string, any>;
+
+    @Column({ type: "varchar", name: "test_done_by", nullable: true })
+    testDoneBy!: string | null
+
+    @Column({ type: "varchar", name: "result_prepared_by", nullable: false })
+    resultPreparedBy!: string;
+
+    @Column({ type: "timestamptz", name: "date_analyzed", nullable: false })
+    dateAnalyzed!: Date;
+
+    @ManyToOne ( () => TestSample, (testSample) => testSample.testResults, { 
+        nullable: false, onDelete: "CASCADE", onUpdate: "CASCADE" })
+    @JoinColumn({ name: "test_sample_id", referencedColumnName: "id" })
+    testSample!: TestSample;
+
+    @ManyToOne ( () => LabStaff, (labStaff) => labStaff.testResults, { 
```

Comment:
delete this. everything staff should be related to who did a test and who prepared the result